plugins/utils/neogen.nix

```diff
@@ -0,0 +1,172 @@
+
+        It'll try to find the first parent that matches a certain type.
+        For example, if you are inside a function, and called `generate({ type = "func" })`,
+        Neogen will go until the start of the function and start annotating for you.
+
+      '';
+    };
+    generateClass = {
+      command = "class";
+      description = "Generates annotation for class.";
+    };
+    generateFunction = {
+      command = "func";
+      description = "Generates annotation for function.";
+    };
+    generateType = {
+      command = "type";
+      description = "Generates annotation for type.";
+    };
+    generateFile = {
+      command = "file";
+      description = "Generates annotation for file.";
+    };
+  };
+in {
+  options.plugins.neogen =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "neogen";
+      package = helpers.mkPackageOption "neogen" pkgs.vimPlugins.neogen;
+      keymaps = mapAttrs (optionsName: properties: helpers.mkNullOrOption types.str properties.description) keymapDef;
+
+      keymapsSilent = mkOption {
+        type = types.bool;
+        description = "Whether Neogen keymaps should be silent";
+        default = false;
+      };
+
+      inputAfterComment = helpers.defaultNullOpts.mkBool true ''
+        If true, go to annotation after insertion, and change to insert mode
+      '';
+      enablePlaceholders = helpers.defaultNullOpts.mkBool true ''
+        If true, enables placeholders when inserting annotation
+      '';
+      languages = helpers.defaultNullOpts.mkNullable types.attrs "see upstream documentation" ''
+        Configuration for languages.
+        template.annotation_convention (default: check the language default configurations)
+        Change the annotation convention to use with the language.
+
+        template.use_default_comment (default: true) Prepend any template line with the default comment for the filetype
+
+        template.position (fun(node: userdata, type: string):(number,number)?)
+        Provide an absolute position for the annotation. If return values are nil, use default position
+
+        template.append If you want to customize the position of the annotation
+
+        template.append.child_name What child node to use for appending the annotation
+
+        template.append.position (before/after) Relative positioning with child_name
+
+        template.<convention_name> (replace <convention_name> with an annotation convention)
+        Template for an annotation convention. To know more about how to create your own template, go here: https://github.com/danymat/neogen/blob/main/docs/adding-languages.md#default-generator
+
+        Example:
+        ```
+        {
+            csharp = {
+                template = {
+                    annotation_convention = "...";
+                };
+            };
+        }
+        ```
+      '';
+      snippetEngine = helpers.defaultNullOpts.mkStr null ''
```

Comment:
```suggestion
      snippetEngine = helpers.mkNullOrOption types.str ''
```

Reply:
Missed that function, thanks. Btw, is there some documentation of these helper functions, or is the only option just going through the code?

Reply:
Unfortunately, as the number of contributors has been fairly limited, we have not an amazing documentation for internal things.
However, you can have a look at the [CONTRIBUTING.md](https://github.com/pta2002/nixvim/blob/main/CONTRIBUTING.md) file. It gives a few information about the tools and the rules we use to package plugins.